src/views/Project/Details/Frameworks/AddFrameworkModal/index.js

```diff
@@ -0,0 +1,263 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { isTruthy } from '@togglecorp/fujs';
+import Faram, { requiredCondition } from '@togglecorp/faram';
+
+import NonFieldErrors from '#rsci/NonFieldErrors';
+import TextInput from '#rsci/TextInput';
+import TextArea from '#rsci/TextArea';
+import LoadingAnimation from '#rscv/LoadingAnimation';
+import DangerButton from '#rsca/Button/DangerButton';
+import SegmentInput from '#rsci/SegmentInput';
+import PrimaryButton from '#rsca/Button/PrimaryButton';
+import Modal from '#rscv/Modal';
+import ModalBody from '#rscv/Modal/Body';
+import ModalHeader from '#rscv/Modal/Header';
+import Cloak from '#components/general/Cloak';
+import {
+    RequestClient,
+    requestMethods,
+} from '#request';
+
+import notify from '#notify';
+import _ts from '#ts';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    frameworkId: PropTypes.number,
+    // eslint-disable-next-line react/no-unused-prop-types
+    setActiveFramework: PropTypes.func.isRequired,
+    closeModal: PropTypes.func,
+    isClone: PropTypes.bool,
+    // eslint-disable-next-line react/forbid-prop-types
+    frameworkCreateRequest: PropTypes.object,
```

Comment:
These are required fields. Is there a problem with react-rest-request?

Reply:
Sometimes it used to be undefined while mounting, couldn't replicate the issue now.
I will make it required for now.